src/java.base/share/classes/sun/reflect/annotation/AnnotationParser.java

```diff
@@ -123,7 +123,7 @@ private static Map<Class<? extends Annotation>, Annotation> parseAnnotations2(
                 if (AnnotationType.getInstance(klass).retention() == RetentionPolicy.RUNTIME &&
                     result.put(klass, a) != null) {
                         throw new AnnotationFormatError(
-                            "Duplicate annotation for class: "+klass+": " + a);
+                            "Duplicate annotation for class: "+klass+": " + a + ": " + container);
```

Comment:
Good RFE.  I suggest a message like this:
```java
"Duplicate annotation " + klass + " in " + container
```
So it will render something like "Duplicate annotation interface java.lang.Deprecated in class com.example.UserClass"

And the actual annotation content doesn't matter as much - it only jams up the message.
Unfortunately we don't have the location for the annotation, but printing out the class name should suffice mostly.

Reply:
Thanks, change made and resolving conversation...